Sprint-3/quote-generator/quotes.js

```diff
@@ -491,3 +491,57 @@ const quotes = [
 ];
 
 // call pickFromArray with the quotes array to check you get a random quote
+
+function displayRandomQuote() {
+  const quoteElement = document.getElementById("quote");
+  const authorElement = document.getElementById("author");
+  const maninElement = document.querySelector("main");
+
+  // Fade out current content
+  quoteElement.style.opacity = "0";
+  authorElement.style.opacity = "0";
+  maninElement.style.opacity = "0";
+
+  setTimeout(() => {
+    const randomQuote = pickFromArray(quotes);
+
+    quoteElement.textContent = randomQuote.quote;
+    authorElement.textContent = `_ ${randomQuote.author}`;
+
+    quoteElement.style.opacity = "1";
+    authorElement.style.opacity = "1";
+    maninElement.style.opacity = "1";
+  }, 300);
+}
+
+let timer = null;
+
+document.getElementById("toggle-timer").addEventListener("change", (event) => {
+  const statusSpan = document.getElementById("status");
+
+  if (event.target.checked) {
+    statusSpan.textContent = "Auto-Play is :ON";
+
+    timer = setTimeout(() => {
+      timer = setInterval(displayRandomQuote, 60000);
+      displayRandomQuote();
+    }, 6000);
+  } else {
+    statusSpan.textContent = "Auto Change";
+
+    clearTimeout(timer);
```

Comment:
Can also remove this line if you rewrite the code in lines 525-528.

Reply:
Thank you for the suggestion! I’ve considered it, but I can’t remove that line because it’s essential for the current functionality. If I rewrite lines 525-528 as suggested, the specific behavior tied to clearing the timeout would no longer work as intended. Please let me know if you’d like me to elaborate further on the current approach!